main.tf

```diff
@@ -14,28 +14,38 @@
  * limitations under the License.
  */
 
+locals {
+  tables = { for table in var.tables : table["table_id"] => table }
```

Comment:
Can we not just convert `var.tables` to a set when used with `for_each`? This seems like unnecessary complexity just to refer to `table_id` as the key.

Reply:
`for_each` accepts a map or a set of strings, but `var.tables` is a list of objects, so converting it to a set it not going to be something supported b y `for_each`